app/util.js

```diff
@@ -0,0 +1,84 @@
+'use strict';
+
+const request = require('request-promise');
+
+// returns a Date representing tomorrow at 12pm so that
+// we can issue darksky 'time machine' requests to get an
+// idea of the weather tomorrow at a time that should be
+// relevant to someone interested in the weather
+function getTomorrowAtNoon() {
+    let tomorrow = new Date();
+    tomorrow.setDate(tomorrow.getDate() + 1);
+    tomorrow.setHours(12);
+    return tomorrow;
+}
+
+// fixed coordinates used to query darksky
+const latitude = 38.8892681;
+const longitude = -77.0501425;
+const darkskyToken = process.env.DARKSKY_TOKEN || '';
+const darkskyUrlBase = `https://api.darksky.net/forecast/${darkskyToken}/${latitude},${longitude}`;
+
+// returns a promise that resolves to the JSON representing the current weather
+// report from darksky on success
+function getCurrentWeatherReport() {
+    // we don't need the minutely or hourly forecast, so we'll exclude them
+    const darkskyCurrentlyUrl = `${darkskyUrlBase}` + '?exclude=minutely,hourly';
+    return request({ uri: darkskyCurrentlyUrl, json: true });
+}
+
+function getTomorrowsWeatherReport() {
+    const darkskyTomorrowUrl = `${darkskyUrlBase}` + getTomorrowAtNoon().getTime() + '?exclude=currently,minutely,hourly';
+    return request({ uri: darkskyTomorrowUrl, json: true });
+}
+
+function generateCurrentSummary(dataBlock) {
+    return generateWeatherReport(dataBlock, 'is currently', 'feels like');
+}
+
+function generateTomorrowsSummary(dataBlock) {
+    return generateWeatherReport(dataBlock, 'tomorrow will be', 'will feel like');
+}
+
+// maps the darksky 'icon' field to an appropriate emoji representation!
+// not supported by jshint yet - https://github.com/jshint/jshint/pull/2413
+const icon2Emoji = {
```

Comment:
Slack emojis and the Darksky `icon` field led me to attempt mapping the icon field which has a fixed set of values to an appropriate weather emoji :cloud_with_lightning:

Not all weather conditions had an appropriate emoji, and I quickly found out some emojis require more than one character. Then I learned ECMAScript 6 added support for specifying any Unicode code point with one symbol!

http://2ality.com/2015/01/es6-strings.html